wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceValidator.java

```diff
@@ -69,23 +68,12 @@ public void validateConfig(WorkspaceConfig config) throws ValidationException, S
             + "latin letters, underscores, dots, dashes and must start and end only with digits, "
             + "latin letters or underscores");
 
-    // environments
-    check(!isNullOrEmpty(config.getDefaultEnv()), "Workspace default environment name required");
-    checkNotNull(config.getEnvironments(), "Workspace must contain at least one environment");
-    check(
-        config.getEnvironments().containsKey(config.getDefaultEnv()),
-        "Workspace default environment configuration required");
-
-    for (Environment environment : config.getEnvironments().values()) {
-      checkNotNull(environment, "Environment must not be null");
-      Recipe recipe = environment.getRecipe();
-      checkNotNull(recipe, "Environment recipe must not be null");
-      checkNotNull(recipe.getType(), "Environment recipe type must not be null");
-
-      for (Entry<String, ? extends MachineConfig> machineEntry :
-          environment.getMachines().entrySet()) {
-        validateMachine(machineEntry.getKey(), machineEntry.getValue());
-      }
+    // if environments/default environment name are empty and it is sidecar-based workspace
+    // then do not check env since it is a workspace with tooling only
+    if ((config.getEnvironments() != null && !config.getEnvironments().isEmpty())
```

Comment:
Maybe it would be simpler if you invoke `validateEnvironments(config);` here always, and inside it check
```
  if (environmentIsConfigured && isSidecarWorkspace) {
    do not check env since it is a workspace with tooling only
    return;
  }
```
WDYT?

Reply:
Yeah, the code might be better in that case. I rework the code a bit to apply this suggestion. 
But it won't that simple because it is OK when both default env and environments are not set.